google-cloud-debugger/README.md

```diff
@@ -1,32 +1,177 @@
 # google-cloud-debugger
 
-[Stackdriver Debugger](https://cloud.google.com/debugger/) ([docs](https://cloud.google.com/debugger/docs/)) lets you inspect the state of a running application at any code location in real time, without stopping or slowing down the application, and without modifying the code to add logging statements. You can use Stackdriver Debugger with any deployment of your application, including test, development, and production. The Ruby debugger adds minimal request latency, typically less than 50ms, and only when the application state is captured. In most cases, this is not noticeable by users.
+[Stackdriver Debugger](https://cloud.google.com/debugger/) lets you inspect the state of a running application at any code location in real time, without stopping or slowing down the application, and without modifying the code to add logging statements. You can use Stackdriver Debugger with any deployment of your application, including test, development, and production. The Ruby debugger adds minimal request latency, typically less than 50ms, and only when the application state is captured. In most cases, this is not noticeable by users.
 
 - [google-cloud-debugger documentation](http://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud-debugger/master/google/cloud/debugger)
 - [google-cloud-debugger on RubyGems](https://rubygems.org/gems/google-cloud-debugger)
 - [Stackdriver Debugger documentation](https://cloud.google.com/debugger/docs/)
 
 ## Quick Start
 
-Setting up Stackdriver Debugger involves three steps:
+Install the gem directly:
 
-1. Add the `google-cloud-debugger` library to your app.
-2. Register your app's source code.
-3. Deploy your app and set a breakpoint.
+```sh
+$ gem install google-cloud-debugger
+```
 
-See the
-[google-cloud-debugger documentation](http://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud-debugger/master/google/cloud/debugger)
-for a quick tutorial.
+Or install through Bundler:
 
-## Authentication
+1. Add the `google-cloud-debugger` gem to your Gemfile:
+
+```ruby
+gem "google-cloud-debugger"
+```
+
+2. Use Bundler to install the gem:
+
+```sh
+$ bundle install
+```
+
+Alternatively, check out the [`stackdriver`](../stackdriver) gem that includes 
+the `google-cloud-debugger` gem.
+
+## Enable Stackdriver Debugger API
+
+The Stackdriver Debugger agent would need the [Stackdriver Debugger 
+API](https://console.cloud.google.com/apis/library/clouddebugger.googleapis.com) 
+to be enabled on your Google Cloud project. Make sure it's enabled if not 
+already.
+
+## Enabling the Debugger agent
+
+If you're using Ruby on Rails and the `stackdriver` gem, they automatically 
```

Comment:
This section is a little confusing because of all the cases (and indeed some cases like stackdriver gem/sinatra are not clear). The fact that we're covering both the `stackdriver` gem case and the `google-cloud-debugger` gem case, makes this more complicated than what we have on cloud.google.com.


I suggest breaking this into three sections with headings "Enabling the Debugger agent with Ruby on Rails", "Enabling the Debugger agent with another framework", and "Enabling the Debugger agent without a Rack-based framework".

The Rails case should start with the non-stackdriver-gem case (i.e. `require "google/cloud/debugger/rails"`) since that is the main case we documented above under installation. Then note that if the stackdriver gem is being used, the require may be omitted.

The other two case look fine as is.

Reply:
Also these headers should probably be level-3 (###)

Reply:
Updated.